### PR TITLE
Fully symbolic memory, multiplex entry points, fair searcher

### PR DESCRIPTION
**Pull request by misonijnik**

### CODE REVIEW

File: lib/Core/Searcher.cpp [hunk 1]
```diff
@@ -781,3 +784,145 @@ void InterleavedSearcher::printName(llvm::raw_ostream &os) {
...
+  for (auto function : localFunction) {
+
+    ExecutionState *currTState =
+        currFunction.count(function) != 0 ? current : nullptr;
+
+    if (!isThereKFunction(function)) {
+      addKFunction(function);
+    }
+
+    searchers.at(function)->update(currTState, addedKStates[function],
```
Comment (S1eGa) [inline lib/Core/Searcher.cpp:863]: `at` is a checked version of `[]`. May has a little affect on performance

_(In `update` below you use `[]` :))_

File: test/Feature/DefineFixedObject.c [hunk 2]
```diff
@@ -2,15 +2,16 @@
...
 #define ADDRESS ((int *)0x0080)
 
 int main() {
-  klee_define_fixed_object(ADDRESS, 4);
-
-  int *p = ADDRESS;
-
+  int *p = klee_define_fixed_object(ADDRESS, 4);
+  klee_print_expr("p", p);
+  klee_print_expr("ADDRESS", ADDRESS);
```
Comment (S1eGa) [inline test/Feature/DefineFixedObject.c:13]: Forgot to remove prints?

File: lib/Expr/Expr.cpp [hunk 3]
```diff
@@ -2567,3 +2861,141 @@ ref<Expr> IsSubnormalExpr::either(const ref<Expr> &e0, const ref<Expr> &e1) {
...
+
+ref<Expr> PointerExpr::Not() {
+  return PointerExpr::create(getBase(), NotExpr::create(getValue()));
+}
+
+ref<Expr> PointerExpr::Eq(const ref<PointerExpr> &RHS) {
+  ref<Expr> areValuesEq = EqExpr::create(getValue(), RHS->getValue());
+  ref<Expr> areBasesEq = EqExpr::create(getBase(), RHS->getBase());
+  // ref<Expr> areBasesNotNull = NotExpr::create(OrExpr::create(
+  //     Expr::createIsZero(getBase()), Expr::createIsZero(RHS->getBase())));
```
Comment (S1eGa) [inline lib/Expr/Expr.cpp:2962]: Please, do not leave commented code.

File: runtime/Runtest/intrinsics.c [hunk 4]
```diff
@@ -69,8 +69,9 @@ void recursively_allocate(KTestObject *obj, size_t index, void *addr,
                            obj->pointers[i].index, 0, 1);
     }
     void *offset_addr = (void *)(addresses[index] + (obj->pointers[i].offset));
-    memcpy(offset_addr, &addresses[obj->pointers[i].index], sizeof(void *));
```
Comment (S1eGa) [inline runtime/Runtest/intrinsics.c:72]: Likely it is okay. Just to be sure: was it a bug?

**Review by S1eGa — CHANGES_REQUESTED** [review]

File: runtime/Runtest/intrinsics.c (hunk 4 above)
Comment (misonijnik) [reply runtime/Runtest/intrinsics.c:72]: Yes, it was.

**Review by S1eGa — APPROVED** [review]

**Pull request merged**